### DISCUSSION TITLE
Payments methods not supported (Klarna and Paypal Checkout)

### DISCUSSION THREAD

**Discussion by usmanali84** [woocommerce/woocommerce#45085] [WooCommerce, Blocks, and Site Editing]
## What do you think is missing from the Cart & Checkout blocks?
Hi,
please fix the payment methods. We use Stripe, Paypal checkout and Klarna payment gateway. Just stripe as a payment seems to be working so we had to remove block as we need other payments methods to work as well.
I hope you will add the other payment methods soon as we will love to use your blocks for Cart and checkout.
Regards
Usman Khan

**Comment by Aljullu** [comment]
Hi there @usmanali84 👋

Thanks for taking the time to make us aware that PayPal Checkout and Klarna aren't working with the Checkout and Cart blocks. These blocks are still in early access (via the feature plugin) and it's _expected_ that some existing payment methods will not work with the blocks.

Currently it's up to payment method extension developers to add support for the new checkout flow offered via the cart and checkout blocks. We have some [documentation on that here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/docs/extensibility/payment-method-integration.md) and will be publicising this more in the coming months.

Your best course of action right now if you are using a payment method that isn't working with the blocks is to notify the developer of the extension and ask if they can add support.

Until then you can keep on top of the [status of the Cart & Checkout Blocks in this documentation](https://docs.woocommerce.com/document/cart-checkout-blocks-support-status/).

**Comment by usmanali84** [comment]
Hi Albert,

Thank you for getting back to me.
I have tried to reach out to klarna developer team and the only thing I
have found is this:
WooCommerce Blocks
<https://woocommerce.com/products/woocommerce-gutenberg-products-block/>

Error: I can not see the Klarna Checkout payment way in the checkout and
I'm using *WooCommerce Blocks *and their *Checkout Design Widget*.
Solution: WooCommerce Blocks got a *Checkout Design Widget* that is causing
this as it is ONLY compatible with these payment gateways
<https://docs.woocommerce.com/document/cart-checkout-blocks-support-status/>.
The
solution is to use the *shortcode widget* instead and use the standard
WooCommerce shortcode [woocommerce_checkout]
And I don't even know how to reach out to the Paypal developer team and
Paypal is the most important payment method as most of the customers pay by
using Paypal. I hope you will add these payment methods soon as we will
love to use Woocommerce Blocks for Cart and Checkout.

Regards
Usman Khan

**Comment by Aljullu** [comment]
Hi @usmanali84, thanks for contacting them. We are working on incrementing the number of payment gateways that support the Cart and Checkout blocks, but some of this work depends on their developers, so it's out of our hands. Hopefully as more users demand it they will prioritize the work.

Unfortunately, I can't give you any expected time when PayPal Checkout and Klarna will be supported. As they reported, currently the only solution is to use the [Checkout shortcode](https://docs.woocommerce.com/document/woocommerce-shortcodes/#section-6) until those payment methods add support for the Cart and Checkout.

**Comment by thisissandip** [comment]
Hey @usmanali84,

I am following up to let you know that we have now integrated Klarna Payments: https://woocommerce.com/products/klarna-payments/ with WooCommerce Blocks.

You can find the compatible payment gateway and WooCommerce Blocks version on our documentation here: https://woocommerce.com/document/cart-checkout-blocks-support-status/#payment-gateways

I hope this helps.

Please feel free to contact us here: https://wordpress.org/support/plugin/woo-gutenberg-products-block/ if you have any further questions.

We will be happy to assist you!
